src/main/java/org/osc/controller/nsc/entities/InspectionHookNSCEntity.java

```diff
@@ -0,0 +1,79 @@
+package org.osc.controller.nsc.entities;
+
+import static javax.persistence.FetchType.LAZY;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+
+@Entity
+@Table(name="InspectionHook")
+public class InspectionHookNSCEntity {
+	
+	private String hookId;
+	private NetworkElementNSCEntity inspectedPort;
+    private InspectionPortNSCEntity inspectionPort; 
+    private Long tag;
+    private Long hookOrder;
+    private String encType;
+    private String failurePolicyType;
+	
+    public InspectionHookNSCEntity() {}
+    
+	
+	@Id
+	public String getHookId() {
+		return hookId;
+	}
+	public void setHookId(String hookId) {
+		this.hookId = hookId;
+	}
+	
+	@OneToOne(cascade=CascadeType.ALL, orphanRemoval=false, fetch=LAZY, optional=true, mappedBy="inspectionHook")
+	@JoinColumn(name="inspectedPortId", nullable=true, unique=true, updatable=true)
+	public NetworkElementNSCEntity getInspectedPort() {
+		return inspectedPort;
+	}
+	public void setInspectedPort(NetworkElementNSCEntity inspectedPort) {
+		this.inspectedPort = inspectedPort;
+	}
+
+	@OneToOne(cascade=CascadeType.ALL, orphanRemoval=true, fetch=LAZY, optional=true)
+	@JoinColumn(name="inspectionPortId", nullable=true, unique=true, updatable=true)
+	public InspectionPortNSCEntity getInspectionPort() {
```

Comment:
I dont think this should be onetoone, i believe the same inspectionport can be used by different inspectionhooks,